ext/ffi/lib.rs

```diff
@@ -948,6 +997,32 @@ where
           return Err(type_error("Invalid FFI pointer type, expected null, BigInt, ArrayBuffer, or ArrayBufferView"));
         }
       }
+      NativeType::Struct(_) => {
+        if let Ok(value) = v8::Local::<v8::ArrayBufferView>::try_from(value) {
+          let byte_offset = value.byte_offset();
+          let backing_store = value
+            .buffer(scope)
+            .ok_or_else(|| {
+              type_error(
+                "Invalid FFI ArrayBufferView, expected data in the buffer",
+              )
+            })?
+            .get_backing_store();
+          let pointer = if byte_offset > 0 {
+            &backing_store[byte_offset..] as *const _ as *const u8
+          } else {
+            &backing_store[..] as *const _ as *const u8
+          };
```

Comment:
I belive that `byte_offset` is unsigned (`usize`)? If so, the check is tautological?
```suggestion
          let pointer = backing_store[byte_offset..].as_ptr().cast::<u8>();
```

Reply:
Nope. It's checking for non-zeroness. And there's a definite downside to using `backing_store[byte_offset..]`: It will panic if the backing store has a size of 0.

This is because in Rust that will mean "take the zeroth element, and all elements after that from this slice", but for a zero element vector there is no zeroth element and thus we're accessing the vector beyond its limits.

Reply:
Ah! Understood, make sense, but it's not obvious at a first glance? Maybe a comment?

Reply:
Comment would not be out of place, though it'd be copied into quite a few places.

Reply:
> Nope. It's checking for non-zeroness. And there's a definite downside to using `backing_store[byte_offset..]`: It will panic if the backing store has a size of 0.

(Mentioned in the other thread about the latter, but as for non-zeroness, it would have to be `<=`)

Reply:
`byte_offset > 0` literally says: "If byte_offset is larger than 0". ie. If there is an offset, then do slice-reference starting from that offset, otherwise just slice-reference the entire buffer.

Reply:
(My bad! I err)